### PR TITLE
fix: refresh onboarding item when context value gets updated (#4597)

### LINKED ISSUE #4597: Rerunning requirements check for Onboarding Podman workflow does not reflect actual state
### Bug description

I got into a requirement check form/page on Windows during podman onboarding workflow execution.
I was asked to:
* enable virtualization platform
* set WSL to version 2
* install WSL
I have enabled Virtualization Platform and pressed `Rerun requirements check` button.

It still wanted me to enable  Virtualization Platform. So I guess that actual check functionality did not take into account changes.

### Operating system

Windws 10

### Installation Method

Installer from website/GitHub releases

### Version

next (development version)

### Steps to reproduce

0. Start with a fresh machine
1. Install Podman Desktop
2. On dashboard, go to setup - aka onboarding
3. Let the requirements check to verify actual state of the host
4. Assert: WSL not installed, VP (Virtualization Platform) not enabled
5. Enable VP - see the link for details
6. Rerun requirements check
7. Assert: VP now passes, WSL still not installed
Actual: VP enabled is still red, need to restart podman desktop or force reload to get proper status in the requirements check

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

it allows an onboarding item to be refreshed when the context value gets updated

### Screenshot / video of UI

onboarding_refresh

### What issues does this PR fix or reference?

it resolves #4597 

### How to test this PR?

1) have the virtualization platform disabled
2) run podman onboarding and see the requirements step fails
3) enable virtualization platform
4) click on rerun requirements check
5) the virtual platform check should be successful 

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by benoitf — APPROVED** [review]
codewise LGTM

File: packages/renderer/src/lib/onboarding/OnboardingItem.svelte [hunk 1]
```diff
@@ -39,22 +43,36 @@ onMount(() => {
...
+  });
+
+  updateHtml();
 });
 
+onDestroy(() => {
+  contextsUnsubscribe?.();
+});
+
+function updateHtml() {
```
Comment (benoitf) [inline packages/renderer/src/lib/onboarding/OnboardingItem.svelte]:
```suggestion
function updateHtml(): void {
```

it is missing return type

**Review by jeffmaury — APPROVED** [review]
LGTM

**Review by cdrage — APPROVED** [review]
Tested with Compose and Podman. Going through code and it also LGTM.

**Pull request merged**